src/dsp_tools/commands/xmlupload/models/ontology_diagnose_models.py

```diff
@@ -81,29 +90,42 @@ def _get_problems_as_df(self) -> pd.DataFrame:
         unpacked: list[dict[str, str]] = list(itertools.chain(*problems))
         return pd.DataFrame.from_records(unpacked)
 
+    def _calculate_num_resources(self, to_count: list[tuple[str, list[str], str]]) -> int:
+        return sum((len(x[1]) for x in to_count))
+
     def _compose_problem_string_cls(self) -> str | None:
         if self.classes:
-            separator = "\n----------------------------\n"
+            if self._calculate_num_resources(self.classes) > 20:
```

Comment:
It's error prone to have a magic value here. (For magic values, see here: https://docs.astral.sh/ruff/rules/magic-value-comparison/)

And in this case it's especially dangerous, because if you change the 20 on line 60, you might easily forget to change it here. So I would store this in a class variable.